lib/reporters/checkstyle.js

```diff
@@ -0,0 +1,76 @@
+/*
+ * Author: Josh Hagins
+ * https://github.com/jawshooah
+ *
+ * Modified from the original by: Boy Baukema
+ * https://github.com/relaxnow
+ */
+
+'use strict';
+
+module.exports = {
+  reporter: function(results) {
+    var files = {},
+      out = [],
+      pairs = {
+        "&": "&amp;",
+        '"': "&quot;",
+        "'": "&apos;",
+        "<": "&lt;",
+        ">": "&gt;"
+      },
+      fileName, i, issue, severity, errorMessage;
+
+    function encode(s) {
+      for (var r in pairs) {
+        if (typeof(s) !== "undefined") {
+          s = s.replace(new RegExp(r, "g"), pairs[r]);
+        }
+      }
+      return s || "";
+    }
+
+    results.forEach(function(result) {
+      // Register the file
+      result.file = result.file.replace(/^\.\//, '');
+      if (!files[result.file]) {
+        files[result.file] = [];
+      }
+
+      // Add the error
+      files[result.file].push({
+        severity: result.type,
+        line: result.lastLine,
+        column: result.lastColumn,
+        message: result.message,
+        source: result.message  // vnu does not output error codes
```

Comment:
Can we leave that property as `null` or whatever might be appropriate? Just duplicating information doesn't seem useful.